sf-fx-runtime-java-runtime/src/main/java/com/salesforce/functions/jvm/runtime/commands/AbstractDetectorCommandImpl.java

```diff
@@ -58,10 +59,35 @@ public final Integer call() throws Exception {
         projectPath,
         projectDetectDuration);
 
+    LOGGER.info("Reading project metadata at path {}...", projectPath);
+    ProjectMetadata projectMetadata =
+        ProjectMetadataParser.parse(projectPath.resolve("project.toml"))
+            .orElseThrow(
+                () -> new IllegalStateException("Cannot parse project metadata (project.toml)!"));
+
+    String salesforceApiVersion = Constants.DEFAULT_SALESFORCE_API_VERSION;
+
+    if (projectMetadata.getSalesforceApiVersion().isPresent()) {
+      salesforceApiVersion = projectMetadata.getSalesforceApiVersion().get();
+    } else {
+      LOGGER.warn(
+          "Project Salesforce API version isn't explicitly defined in project.toml. The default version {} will be used.",
+          Constants.DEFAULT_SALESFORCE_API_VERSION);
+    }
+
+    if (Constants.SUPPORTED_SALESFORCE_API_VERSIONS.contains(salesforceApiVersion)) {
+      LOGGER.info("Project uses Salesforce API version {}.", salesforceApiVersion);
+    } else {
+      LOGGER.error(
+          "Project declares to use unsupported Salesforce API version {}. Exiting.",
```

Comment:
It feels like there is a missing "an" between "use" and "unsupported", but also "Project declares" didn't seem quite right at first read through. 

How about something like this? (Feel free to try an alternative instead)

```suggestion
          "Project configuration specifies an unsupported Salesforce API version {}. Exiting.",
```

Reply:
Always happy to get suggestions for better wording <3 Thanks!